righttyper/righttyper_tool.py

```diff
@@ -1,39 +1,49 @@
 import signal
 import sys
-from types import CodeType, FrameType
-from typing import Any
+from types import CodeType
+from typing import Any, Final
 from collections.abc import Callable
+import functools
 
-from righttyper.righttyper_utils import TOOL_ID, TOOL_NAME
+TOOL_NAME: Final[str] = "righttyper"
+
+def _setup_tool_id(tool_id: int) -> int:
+    while True:
```

Comment:
There is a maximum number of tool IDs (5 atm): see https://docs.python.org/3/library/sys.monitoring.html#tool-identifiers.

Reply:
In other words, the logic won't work if it gets to 6; it will I think enter an infinite loop.